code/modules/mob/living/say.dm

```diff
@@ -263,12 +257,26 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 		eavesdropping = stars(message)
 		eavesrendered = compose_message(src, message_language, eavesdropping, , spans, message_mods)
 
+	var/list/show_overhead_message_to = list()
+	var/list/show_overhead_message_to_eavesdrop = list()
 	var/rendered = compose_message(src, message_language, message, , spans, message_mods)
 	for(var/atom/movable/AM as() in listening)
 		if(eavesdrop_range && get_dist(source, AM) > message_range && !(the_dead[AM]))
+			if(ismob(AM))
```

Comment:
Move this before eavesdrop check and just continue if thing isn't mob and shouldn't see message

Reply:
That would change how it current works though and would make it so intercoms and other devices that hear but aren't people cannot hear whispering.

Reply:
Yes but you're duplicating code here

Reply:
its not duplicated, it adds to a different list

Reply:
:weary: